### DISCUSSION TITLE
Use or-tools in UIPath

### DISCUSSION THREAD

**Discussion by wannestal** [google/or-tools#3810] [Build and configuration issues]
**What version of OR-Tools and what language are you using?**
Version: v9.6/
Language: C++/Java/Python/C#
C#
**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
GLOB (while testing), will use also other solvers
**What operating system (Linux, Windows, ...) and version?**
Windows 10 and 11 professional
**What did you do?**
1. Create new C# project in UIPath
2. Install package Or tools Win x64 by Laurent Perron
3. Import namespace Google.OrTools.Lineairsolver
4. Use an Invoke Code activity
5. Use the sample from Google (bit adapted to fit UIPath):
   
        // [START solver]
        // Create the linear solver with the GLOP backend.
        Solver solver = Solver.CreateSolver("GLOP");
		Console.WriteLine("START");
        if (solver is null)
        {
			Console.WriteLine("solver null");
            return;
        }
        // [END solver]

        // [START variables]
        // Create the variables x and y.
        Google.OrTools.LinearSolver.Variable x = solver.MakeNumVar(0.0, 1.0, "x");
        Google.OrTools.LinearSolver.Variable y = solver.MakeNumVar(0.0, 2.0, "y");

        Console.WriteLine("Number of variables = " + solver.NumVariables());
        // [END variables]

        // [START constraints]
        // Create a linear constraint, 0 <= x + y <= 2.
        Google.OrTools.LinearSolver.Constraint ct = solver.MakeConstraint(0.0, 2.0, "ct");
        ct.SetCoefficient(x, 1);
        ct.SetCoefficient(y, 1);

        Console.WriteLine("Number of constraints = " + solver.NumConstraints());
        // [END constraints]

        // [START objective]
        // Create the objective function, 3 * x + y.
        Objective objective = solver.Objective();
        objective.SetCoefficient(x, 3);
        objective.SetCoefficient(y, 1);
        objective.SetMaximization();
        // [END objective]

        // [START solve]
        solver.Solve();
        // [END solve]

        // [START print_solution]
        Console.WriteLine("Solution:");
        Console.WriteLine("Objective value = " + solver.Objective().Value());
        Console.WriteLine("x = " + x.SolutionValue());
        Console.WriteLine("y = " + y.SolutionValue());
        // [END print_solution]

**What did you expect to see**
The result of the code, printed in the console
**What did you see instead?**
Error: MISSING dll google-ortools-native.dll
Make sure you include information that can help us debug (full error message, model Proto).

**Anything else we should know about your project / environment**
After trying proposed solutions on the internet, I've decided to place google-ortools-native.dll in de project folder of the UIPath robot. Then it worked. 
So why is the dll not in the project folder after installing the package from NuGet? With my solution, it is not reliable for future updates.

**Comment by lperron** [comment]
We only support basic C#, no external frameworks. Moving to discussions.

**Reply by wannestal** [reply]
If someone has a suggestion for a workaround to call the solver from a UIPath bot...
